web/ui/src/Services/AppsController.js

```diff
@@ -95,32 +95,26 @@
 
             service.model.Endpoints.forEach(endpoint => {
                 if(endpoint.VHostList){
-                    endpoint.VHostList.forEach(vHost => endPoints.push(vHost));
+                    endpoint.VHostList.forEach(vHost => {
+                        vHost.URL = utils.publicEndpointURL($scope, vHost);
+                        endPoints.push(vHost);
+                    });
                 }
                 if(endpoint.PortList){
-                    endpoint.PortList.forEach(port => endPoints.push(port));
+                    endpoint.PortList.forEach(port => {
+                        port.URL = utils.publicEndpointURL($scope, port);
+                        endPoints.push(port);
+                    });
                 }
             });
 
-            endPoints.sort();
+            endPoints.sort(function(pep1, pep2) { return pep1.URL.localeCompare(pep2.URL); });
```

Comment:
not strictly necessary, but es6 arrow functions are more succint for this exactly usage:
    endPoints.sort((pep1, pep2) =? pep1.URL.localeCompare(pep2.URL));